src/RootCloseWrapper.js

```diff
@@ -54,7 +60,22 @@ export default class RootCloseWrapper extends React.Component {
   }
 
   render() {
-    return React.Children.only(this.props.children);
+    // Wrap the child in a new element, so the child won't have to handle
+    // potentially combining multiple onClick listeners.
+    const child = React.cloneElement(
+      React.Children.only(this.props.children),
+      {ref: component => this._wrappedElement = component}
+    );
+
+    return (
+      <div onClick={suppressRootClose}>
+        {child}
```

Comment:
Does adding an extra element potentially mess with layout? What would be the downside with just chaining the onClick? Or if that fails maybe just adding a native handler with addEventListener?

The other concern is that is that will replace any refs the child already has, which is not unlikely if someone needs to call any methods, or reference there overlay component. One alternative is to use the chained function helper to there, that will work if there is an existing ref of the new style, but not if the ref is a string.